aries_cloudagent/utils/http.py

```diff
@@ -96,9 +101,65 @@ async def fetch(
                     response: ClientResponse = await session.get(url, headers=headers)
                     if response.status < 200 or response.status >= 300:
                         raise ClientError(
-                            f"Bad response from server: {response.status}"
+                            f"Bad response from server: {response.status} - "
+                            f"{response.reason}"
                         )
                     return await (response.json() if json else response.text())
             except (ClientError, asyncio.TimeoutError) as e:
                 if attempt.final:
                     raise FetchError("Exceeded maximum fetch attempts") from e
+
+
+async def put(
+    url: str,
+    file_data: dict,
+    extra_data: dict,
+    *,
+    retry: bool = True,
+    max_attempts: int = 5,
+    interval: float = 1.0,
+    backoff: float = 0.25,
```

Comment:
Same as previous comment: they don't appear to be used anywhere?

Reply:
Sure they are: see below

Reply:
Ha, I missed that line. Thanks.